### PR TITLE
fix: handle `baseUrl` suffix for GHES

### LINKED ISSUE #185: Setting baseUrl for GHES results in 406 when using with @octokit/auth-app
This is a follow up issue to https://github.com/octokit/auth-app.js/issues/111 @lencioni 

Minimal test case to reproduce the problem

```js
const { createAppAuth } = require("@octokit/auth-app");
const { request } = require("@octokit/request");
const { graphql } = require("@octokit/graphql");

const GITHUB_BASE_URL = "https://ghe.io";

test();

function test() {
  const auth = createAppAuth({
    id: process.env.APP_ID,
    privateKey: process.env.PRIVATE_KEY,
    installationId: process.env.INSTALLATION_ID,
  });
  const graphqlWithAuth = graphql.defaults({
    baseUrl: `${process.env.HOST}/api`,
    request: {
      hook: auth.hook,
    },
  });

  graphqlWithAuth(
    `query {
      viewer {
        login
      }
    }`
  ).then(console.log, (error) => {
    console.log(error);
    process.exit(1);
  });
}
```

The script currently fails with `406`

The problem is that using `auth.hook` will utilize the request instance that it was passed into. That means that the `@octokit/auth-app` instance will use `${process.env.HOST}/api` as baseURL which will result it sending the request to create an installation access token to

```
POST https://[GHES HOST]/api/app/installations/83/access_tokens
```

But the correct URL is

```
POST https://[GHES HOST]/api/v3/app/installations/83/access_tokens
```

I will workaround it by updating the README to set baseUrl to `https://[GHES host]/api/v3` instead of  `https://[GHES host]/api`, and then replacing the `/api/v3` suffix in the baseUrl with just `/api` for the GraphQL requests only.

This will resolve the problem at hand. The real fix to this is to remove any suffix from the `baseUrl` option across all Octokit packages, and add it only as needed. We have been bitten by this several times in the past. This will require much more work and might introduce breaking changes, so I'll address that later.

### PR DESCRIPTION
**Pull request by gr2m**
fixes #185
fixes https://github.com/octokit/auth-app.js/issues/111

/cc @lencioni

### CODE REVIEW

File: test/defaults.test.ts [hunk 1]
```diff
@@ -121,4 +121,30 @@ describe("graphql.defaults()", () => {
       expect(result).toStrictEqual(mockData);
     });
   });
+
+  it.only("handle baseUrl set with /api/v3 suffix", () => {
```
Comment (lencioni) [inline test/defaults.test.ts:125]:
```suggestion
  it("handle baseUrl set with /api/v3 suffix", () => {
```

File: src/graphql.ts [hunk 2]
```diff
@@ -18,6 +18,8 @@ const NON_VARIABLE_OPTIONS = [
   "mediaType",
 ];
 
+const GHES_V3_SUFFIX_REGEX = /\/api\/v3$/;
```
Comment (lencioni) [inline src/graphql.ts:21]: Should this include an optional trailing slash to be safe?
```suggestion
const GHES_V3_SUFFIX_REGEX = /\/api\/v3\/?$/;
```

**Comment by lencioni** [comment]
Thanks for working on this!

File: src/graphql.ts (hunk 2 above)
Comment (gr2m) [reply src/graphql.ts:21]: won't hurt, thanks!

File: test/defaults.test.ts (hunk 1 above)
Comment (gr2m) [reply test/defaults.test.ts:125]: Ohhh good catch. Interesting that coverage was 100% even when only this test was run 🤔 Maybe I should add a linting rule from preventing `it.only` statements to be merged

**Pull request merged**